### PR TITLE
feat(LH-69947) Create SDC on VMware using Terraform

### PR DESCRIPTION
**Pull request by siddhuwarrier**
- feat(lh-69947): add first cut of terraform-vsphere-cdo-sdc
- ci(lh-69947): add github actions

### CODE REVIEW

File: examples/terraform.tfstate.backup [hunk 1]
```diff
@@ -0,0 +1,154 @@
+{
```
Comment (bobertyy) [inline examples/terraform.tfstate.backup:1]: TF state shouldn't be in this repo

File: examples/terraform_env_vars.sh.sample [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export TF_VAR_vsphere_server=<insert-value>
```
Comment (bobertyy) [inline examples/terraform_env_vars.sh.sample:1]: I don't think we need this file?

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
+resource "vsphere_folder" "sdc_folder" {
+  path          = local.folder_path
+  type          = "vm"
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+resource "vsphere_content_library" "sdc_content_library" {
+  name            = "${var.cdo_tenant_name}-content-library"
+  description     = "Content library to hold SDC OVAs for CDO tenant ${var.cdo_tenant_name}"
+  count           = var.sdc_content_library_item_id == null ? 1 : 0
```
Comment (bobertyy) [inline main.tf:10]: Lets put the count at the top of this resource like:

```hcl
resource "vsphere_content_library" "sdc_content_library" {
  count = var.sdc_content_library_item_id == null ? 1 : 0

  name            = "${var.cdo_tenant_name}-content-library"
  description     = "Content library to hold SDC OVAs for CDO tenant ${var.cdo_tenant_name}"
  storage_backing = [data.vsphere_datastore.datastore.id]
}
```

File: examples/terraform_env_vars.sh.sample (hunk 2 above)
Comment (siddhuwarrier) [reply examples/terraform_env_vars.sh.sample:1]: No we need this file. This is a sample file; if you look at the README, I say you can update this file with the right values and source it to try as an example. I'll make it clearer in the README

File: main.tf (hunk 3 above)
Comment (siddhuwarrier) [reply main.tf:10]: Done.

File: examples/terraform.tfstate.backup (hunk 1 above)
Comment (siddhuwarrier) [reply examples/terraform.tfstate.backup:1]: Removed in my subsequent PR.

**Pull request merged**